alacritty_terminal/src/term/mod.rs

```diff
@@ -417,30 +395,30 @@ impl<T> Term<T> {
         }
     }
 
+    /// Collect the information about the changes in the lines, which
+    /// could be used to minimize the amount of drawing operations.
+    ///
+    /// The user controlled elements, like `Vi` mode cursor and `Selection` are **not** part of the
+    /// collected damage state. Those could easily be tracked by comparing their old and new
+    /// value between adjacent frames.
+    ///
+    /// After reading damage [`reset_damage`] should be called.
+    ///
+    /// [`reset_damage`]: Self::reset_damage
     #[must_use]
-    pub fn damage(&mut self, selection: Option<SelectionRange>) -> TermDamage<'_> {
+    pub fn damage(&mut self) -> TermDamage<'_> {
         // Ensure the entire terminal is damaged after entering insert mode.
         // Leaving is handled in the ansi handler.
         if self.mode.contains(TermMode::INSERT) {
             self.mark_fully_damaged();
         }
 
-        // Update tracking of cursor, selection, and vi mode cursor.
-
-        let display_offset = self.grid().display_offset();
-        let vi_cursor_point = if self.mode.contains(TermMode::VI) {
-            point_to_viewport(display_offset, self.vi_mode_cursor.point)
-        } else {
-            None
-        };
-
         let previous_cursor = mem::replace(&mut self.damage.last_cursor, self.grid.cursor.point);
-        let previous_selection = mem::replace(&mut self.damage.last_selection, selection);
-        let previous_vi_cursor_point =
-            mem::replace(&mut self.damage.last_vi_cursor_point, vi_cursor_point);
 
-        // Early return if the entire terminal is damaged.
-        if self.damage.is_fully_damaged {
+        // NOTE: the terminal damage is tracked from the bottom, however scrolling could damage
+        // the viewport due to lines going off the scrollback, hence we don't check for display
+        // offset when we have full damage.
```

Comment:
I'm not sure this is the right place for this comment. If there's full damage I don't think anyone would assume you'd need to check anything? So the detail provided here might be correct, but it's useless.

Also the "from the bottom" here is confusing. I'd assume what you mean is that terminal damage is tracked only for the viewport. If that's from the bottom or top of the viewport is irrelevant, the important bit is that it's the viewport we care about.

Reply:
I have no idea where to put this comment as well, because it all sounds like I'm explaining how damage works when we have all the parts in place and test covered. This comment is neither `Why` nor `What`, it's just states the fact on how things happened to work.

>I'd assume what you mean is that terminal damage is tracked only for the viewport.

Viewport is what you _view_, which is not the case, because once you've scrolled your viewport is moved to that offset. I mean the area input can write to, but it's not a viewport.